### ISSUE TITLE
Feature: add support to typedi

### ISSUE THREAD

**Issue by afranioce** [L2jLiga/fastify-decorators#395]
[typedi](https://github.com/typestack/typedi) is a power full tool to DI, so I believe that `fastify-decorators` don't need to implement on your core a DI, maybe create an adapter on you config to set a Container like:

```typescript
import { useContainer } from 'fastify-decorators';
import { Container } from 'typedi';

useContainer(Container);
```

> see https://docs.typestack.community/typedi/v/develop/02-basic-usage-guide/07-usage-with-typeorm

**Comment by L2jLiga** [comment]
Hi there, any description would be helpful

**Comment by L2jLiga** [comment]
So, I've started working on this feature. Currently I've created draft PR with changes for v4.
In general I would like to implement some kind of APIs for hooking library to be able create such things like DI via typedi. You can check it here: https://github.com/L2jLiga/fastify-decorators/pull/457

**Comment by L2jLiga** [comment]
Released as v4.0.0-next.0

For upgrade guide see [Migration to v4](https://github.com/L2jLiga/fastify-decorators/blob/v4/docs/Migration%20to%20v4.md) guide

- [`fastify-decorators`](https://www.npmjs.com/package/fastify-decorators/v/4.0.0-next.0)
- [`@fastify-decorators/simple-di`](https://www.npmjs.com/package/@fastify-decorators/simple-di)
- [`@fastify-decorators/typedi`](https://www.npmjs.com/package/@fastify-decorators/typedi)

**Comment by L2jLiga** [comment]
Released in [`v3.13.0`](https://github.com/L2jLiga/fastify-decorators/releases/tag/v3.13.0)  🎉 

you can override how controllers are instantiated via `classLoader` option:

```typescript
import { fastify } from 'fastify'
import { bootstrap } from 'fastify-decorators'
import { Container } from 'typedi'

const app = fastify()

app.register(bootstrap, {
  classLoader: (clazz) => Container.get(clazz)
})
```
